src/probnum/filtsmooth/gaussfiltsmooth/iterated_kalman.py

```diff
@@ -0,0 +1,132 @@
+"""Iterated Gaussian filtering and smoothing."""
+import numpy as np
+
+import probnum.random_variables as pnrv
+
+from .kalman import Kalman
+from .stoppingcriterion import StoppingCriterion
+
+
+class IteratedKalman(Kalman):
+    """Iterated filter/smoother based on posterior linearisation.
+
+    In principle, this is the same as a Kalman filter; however,
+    1. predict() and update() in each filter step may be repeated.
+    2. approximate Gaussian filtering and smoothing can use posterior linearisation.
+
+    There is an additional method: :meth:`iterated_filtsmooth()`,
+    which computes things like MAP estimates.
+    """
+
+    def __init__(self, kalman, stoppingcriterion=None):
+        self.kalman = kalman
+        if stoppingcriterion is None:
+            self.stoppingcriterion = StoppingCriterion()
+        else:
+            self.stoppingcriterion = stoppingcriterion
+        super().__init__(kalman.dynamod, kalman.measmod, kalman.initrv)
+
+    def filter_step(
+        self, start, stop, current_rv, data, previous_posterior=None, **kwargs
+    ):
+        """Filter step of iterated filter.
+
+        Different to Kalman.filter in the sense that there may be multiple iterated updates for
+        a single predict, or multiple iterated predicts for a single update, or multiple iterated predicts
+        and updates in general.
+        This retrieves methods such as the iterated extended Kalman filter.
+        By further specifying a `previous_posterio` (KalmanPosterior), the first of those possibly
+        iterated updates is linearised at the previous posterior estimate.
+        This retrieves methods such as the iterated extended Kalman smoother.
+
+        Parameters
+        ----------
+        start : float
+            Predict FROM this time point.
+        stop : float
+            Predict TO this time point.
+        current_rv : RandomVariable
+            Predict based on this random variable. For instance, this can be the result
+            of a previous call to filter_step.
+        data : array_like
+            Compute the update based on this data.
+        previous_posterior : KalmanPosterior
+            Posterior distribution of a previous smoothing iteration. Optional.
+            If specified, posterior linearisation is applied.
+
+        Returns
+        -------
+        RandomVariable
+            Resulting filter estimate after the single step.
+        dict
+            Additional information provided by predict() and update().
+            Contains keys `pred_rv`, `info_pred`, `meas_rv`, `info_upd`.
```

Comment:
The alternative I see here is to merge the info dict with `info_pred` and `info_upd`. 
What do you think?

Reply:
Tbh, I dont know... I was thinking about that, too, but then decided against it. The reason being that this way, it is easy to access the single elements (RV or dict) the way that the sub-methods spit them out without doing some weird merging. Of course a dict of dicts is odd... either way, I think both work, but would slightly (60% to 40%) prefer the dict of dicts over flattening things. What is your opinion?

Reply:
With your comment and looking at the code once more, I think the current implementation is probably better than merging the dicts :+1: